### PR TITLE
Fix deployment with error during cm creation

### PR DESCRIPTION
**Pull request by satr**
Fixed config-map sync - removed lister from config-map getter. 
Added error handling for create desired deployment

### CODE REVIEW

File: pkg/apis/kube/config_maps.go [hunk 1]
```diff
@@ -5,37 +5,33 @@ import (
...
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
 )
 
 // CreateConfigMap Create config map
 func (kubeutil *Kube) CreateConfigMap(namespace string, configMap *corev1.ConfigMap) (*corev1.ConfigMap, error) {
-	return kubeutil.kubeClient.CoreV1().ConfigMaps(namespace).Create(context.TODO(),
+	createdConfigMap, err := kubeutil.kubeClient.CoreV1().ConfigMaps(namespace).Create(context.TODO(),
```
Comment (nilsgstrabo) [inline pkg/apis/kube/config_maps.go:16]: You can just return directly. Are there situations where Get returns an error and a configmap, or vice versa; nil error and nil configmap?

File: pkg/apis/kube/config_maps.go [hunk 2]
```diff
@@ -5,37 +5,33 @@ import (
...
-		configMap, err = kubeutil.kubeClient.CoreV1().ConfigMaps(namespace).Get(context.TODO(), name, metav1.GetOptions{})
-		if err != nil {
-			return nil, err
-		}
-		log.Debugf("Created environment variables ConfigMap  '%s'", configMap.GetName())
+	configMap, err = kubeutil.kubeClient.CoreV1().ConfigMaps(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
 	}
+	log.Debugf("Created environment variables ConfigMap  '%s'", configMap.GetName())
```
Comment (nilsgstrabo) [inline pkg/apis/kube/config_maps.go:34]: Debug message (Create) does not match operation (Get).
Guess you can return directly from the call to Get. Ref comment above

**Review by nilsgstrabo — CHANGES_REQUESTED** [review]
A few comments related to Get and Create configmap methods

**Review by nilsgstrabo — APPROVED** [review]
A few minor comments, but nothing important

File: pkg/apis/kube/config_maps.go (hunk 1 above)
Comment (satr) [reply pkg/apis/kube/config_maps.go:16]: Fixed

File: pkg/apis/kube/config_maps.go (hunk 2 above)
Comment (satr) [reply pkg/apis/kube/config_maps.go:34]: Removed

**Pull request merged**